File: 61-rotate-list/rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if(head==null) return head;
        int n=1;
        ListNode temp=head;
        while(temp.next!=null){
            temp=temp.next;
            n++;
        }
        k=k%n;
        int t=n-k;
        ListNode temp2=head;
        for(int i=1;i<t;i++){
            temp2=temp2.next;
        }
        temp.next=head;
        head=temp2.next;
        temp2.next=null;
        return head;
    }
}

File: 61-rotate-list/rotate-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        int n=1;
        if(head==nullptr) return head;
        ListNode* temp=head;
        while(temp->next!=nullptr){
            n++;
            temp=temp->next;
        }
        k=k%n;
        int t=n-k;
        ListNode* temp2=head;
        for(int i=1;i<t;i++){
            temp2=temp2->next;
        }
        temp->next=head;
        ListNode* headt=head;
        head=temp2->next;
        temp2->next=nullptr;
        return head;
    }
};